controllers/storagecluster/storagecluster_controller.go

```diff
@@ -225,7 +225,7 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&appsv1.Deployment{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Owns(&corev1.Service{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
 		Owns(&corev1.ConfigMap{}, builder.MatchEveryOwner, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
-		Owns(&corev1.Secret{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Owns(&corev1.Secret{}, builder.MatchEveryOwner, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
Why do we need a MatchEveryOwner here? Isn't StorageCluster the controller owner of the secret we want to watch?

Reply:
controller owner reference set to secrets, ack

Reply:
This comment was not addressed as tall!  You are matching by controller ownership, it does not make sense to match every owner

Reply:
Owner reference were set during creation of secrets in on-boarding job, when the rotate signing key is clicked the re-consiliation is not happening, The onboarding-token needs to be created which is failing. 

Any suggestion for this issue.

Reply:
If there is a controller owner ref on the secret and then someone deletes that secret then a delete event will be sent here. And because the item had an owner ref (with controller set) then a reconcile request will be queued. Now if that is not happening there can be a bug in a couple of places:
- The predicate might be the one the drops the event (should not, but might be)
- The reconcile is initiated but the reconcile code might not re-create the secret (might happen because of stale cache)
- Are we sure we are adding a controller reference on the secret? if not and we are only adding an owner ref then that is your bug

@mrudraia1 You need to identify the issue, and only after you fully understand what is going on you can suggest a fix.

Reply:
StorageCluster is the owner of the secrets, not the controller owner [here](https://github.com/red-hat-storage/ocs-operator/blob/main/onboarding-validation-keys-generator/main.go#L84)

Reply:
> Owner reference were set during creation of secrets in on-boarding job, when the rotate signing key is clicked the re-consiliation is not happening, The onboarding-token needs to be created which is failing.
- ouch, so the issue all along is storagecluster not getting a reconcile event when rotate keys is invoked (ie, deletion of secrets) but not the delay b/n rotation and kubelet updating the secret as diagnosed in https://bugzilla.redhat.com/show_bug.cgi?id=2311012#c5?

@rchikatw was there a discussion w/ @mrudraia1 along these lines when he started to work on this? @mrudraia1 did you observe the same behavior from when the work started on this PR, end of July and in latest revision I don't see this change which basically means we are just refactoring the code w/o any observable behavior and not fixing the issue.

Reply:
@leelavg earlier secret keys are read from the volume mounts , which is now changed to read directly by reading the secrets. 

OwnerReference of the secrets changed to ControllerReference.

Reply:
yes, know with this PR build I tested, I can observe the 
1) creation of  new on-boarding_token immediately after the clicking of Rotate signing key.
2) earlier I was using MatchEveryOwner to watch the secrets, Know the ControllerOwner is set to watch the resources.

Reply:
I am not aware of any discussion made to change the predicate, And not sure why we are changing this OwnerRef -> ControllerRef: [Here](https://github.com/red-hat-storage/ocs-operator/pull/2827/files#diff-5c4a3e6001667acda632743674c8d1463952f5d5d7d35a23c60db6ed8340dd0aR84) And because of this change you might be seeing job creation is not happening when keys are rotated this is my just a guess. AFAIK I know a job is created when keys are rotated and even QE team tested this.